Common/src/config_structure.cpp

```diff
@@ -1442,6 +1442,8 @@ void CConfig::SetConfig_Options() {
   addUnsignedLongOption("LINEAR_SOLVER_RESTART_FREQUENCY", Linear_Solver_Restart_Frequency, 10);
   /* DESCRIPTION: Relaxation factor for iterative linear smoothers (SMOOTHER_ILU/JACOBI/LU-SGS/LINELET) */
   addDoubleOption("LINEAR_SOLVER_SMOOTHER_RELAXATION", Linear_Solver_Smoother_Relaxation, 1.0);
+  /* DESCRIPTION: Custom number of threads used for additive domain decomposition for ILU and LU_SGS (0 is "auto"). */
+  addUnsignedLongOption("LINEAR_SOLVER_PREC_THREADS", Linear_Solver_Prec_Threads, 0);
```

Comment:
With this PR the ILU and LU_SGS preconditioners can now run with a custom number of threads specified through the config file.
For structural cases (and mesh deformation) it gets harder and harder (more linear iterations) to solve the linear systems as more processors are throw at the problem (which is annoying in multizone problems), being able to independently control the number of preconditioner threads alleviates the problem.

Reply:
Yup it uses the same threads that are used everywhere else.
To answer another of your questions:
> When using the mesh solver in a CFD case (so, single zone), is it possible to use different levels of parallelism for the CFD problem and for the mesh deformation problem? And, if so, is it something worth doing?

Although one could change the number of threads used in all mesh deformation and fea routines very easily, at the moment the mesh deformation runs synchronously with the rest of the code, and so running the discretization parts with fewer threads would not be advantageous.
The area where it might be worth reducing the number of threads is those 2 preconditioners, if and only if:
- The linear solver is not able to reach adequate convergence (or is diverging) due to the preconditioner being too weak (too partitioned).
- The code is running on a very bandwidth-limited machine, in which case reducing threads a bit might give a speedup due to fewer linear iterations being required.

Other areas of the code will not suffer as much from having too many threads, since threads do not increase the number of halo points (communication costs) and allow for "automatic" load balancing.